### PR TITLE
feat: add kind support

### PR DESCRIPTION
**Pull request by colesnodgrass**
- add multiple kubernetes provider support; now supports
    - [kind](https://kind.sigs.k8s.io/)
    - [docker-desktop](https://www.docker.com/products/docker-desktop/)
- default provider to `kind`
    - this decreases the prerequisites further by removing the docker-desktop kubernetes requirement
    - only `docker` is now required to already be installed
- change the default port from `80` to `8000`
- add the ability to change which port will be used
    - pass the `--port` parameter
- add help messages for specific failure scenarios:
    - kubernetes specific error message
    - ingress specific error message
    - port specific error message
- clean up telemetry package 
- add some tests, more are still needed

### CODE REVIEW

File: internal/local/k8s/cluster.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	"os"
+	"path/filepath"
+	"sigs.k8s.io/kind/pkg/cluster"
+	"time"
+)
+
+// Cluster is an interface representing all the actions taken at the cluster level.
+type Cluster interface {
+	// Create a cluster with the provided name.
+	Create(portHTTP int) error
```
Comment (perangel) [inline internal/local/k8s/cluster.go:17]: NIT: if we can find a way (that isn't annoying) to not require the port here, that would be cool, but def not a blocker

File: internal/local/k8s/provider.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	if err := os.MkdirAll(filepath.Dir(kubeconfig), permissions); err != nil {
+		return fmt.Errorf("could not create directory %s: %v", kubeconfig, err)
+	}
+
+	return nil
+}
+
+var (
+	// DockerDesktopProvider represents the docker-desktop provider.
+	DockerDesktopProvider = Provider{
```
Comment (perangel) [inline internal/local/k8s/provider.go:45]: Suggestion: Consider moving these to a `map[string]Provider` this way in any error handling you can and enumerate the valid options

**Review by perangel — APPROVED** [review]

File: internal/local/k8s/cluster.go (hunk 1 above)
Comment (colesnodgrass) [reply internal/local/k8s/cluster.go:17]: Agreed.  Will revisit this in a follow-up PR.

File: internal/local/k8s/provider.go (hunk 2 above)
Comment (colesnodgrass) [reply internal/local/k8s/provider.go:46]: Agreed.  Will revisit this in a follow-up PR.

File: internal/local/docker/docker.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+func New() (*Docker, error) {
+	userHome, err := os.UserHomeDir()
+	if err != nil {
+		return nil, fmt.Errorf("could not determine user home directory: %w", err)
+	}
+
+	var dockerCli *client.Client
+
+	switch runtime.GOOS {
+	case "darwin":
```
[the remark this answers is not in the document]
Comment (colesnodgrass) [reply internal/local/docker/docker.go:48]: I ran into problems with that initially, specific to osx.  I'll revisit this in a separate PR.

**Pull request merged**